aws/resource_aws_dms_endpoint_test.go

```diff
@@ -174,125 +206,165 @@ resource "aws_dms_endpoint" "dms_endpoint" {
+			"Service": "dms.amazonaws.com"
+		},
+		"Effect": "Allow"
+	}
+]
 }
 EOF
 }
 
-resource "aws_iam_role_policy" "dms_dynamodb_access" {
-  name = "tf-test-iam-dynamodb-role-policy-%[1]s"
-  role = "${aws_iam_role.iam_role.name}"
+resource "aws_iam_role_policy" "dms_%[2]s_access" {
+name = "tf-test-iam-%[2]s-role-policy-%[1]s"
+role = "${aws_iam_role.iam_role.name}"
 
-  policy = <<EOF
+policy = <<EOF
 {
 "Version": "2012-10-17",
 "Statement": [
 {
-    "Effect": "Allow",
-    "Action": [
-        "dynamodb:PutItem",
-        "dynamodb:CreateTable",
-        "dynamodb:DescribeTable",
-        "dynamodb:DeleteTable",
-        "dynamodb:DeleteItem",
-        "dynamodb:ListTables"
-    ],
-    "Resource": "*"
+	"Effect": "Allow",
+	"Action": [
+			%[3]s
+	],
+	"Resource": "*"
 }
 ]
 }
 EOF
 }
 
-`, randId)
+`, randId, engineName, actionList)
 }
 
-func dmsEndpointDynamoDbConfigUpdate(randId string) string {
+func dmsEndpointTargetUpdateConfig(randId string, engineName string, actionList string) string {
 	return fmt.Sprintf(`
 resource "aws_dms_endpoint" "dms_endpoint" {
-	endpoint_id = "tf-test-dms-endpoint-%[1]s"
-	endpoint_type = "target"
-	engine_name = "dynamodb"
-	service_access_role = "${aws_iam_role.iam_role.arn}"
-	ssl_mode = "none"
-	tags {
-		Name = "tf-test-dynamodb-endpoint-%[1]s"
-		Update = "updated"
-		Add = "added"
-	}
+endpoint_id = "tf-test-dms-endpoint-%[1]s"
+endpoint_type = "target"
+engine_name = "%[2]s"
+service_access_role = "${aws_iam_role.iam_role.arn}"
+ssl_mode = "none"
+tags {
+	Name = "tf-test-%[2]s-endpoint-%[1]s"
+	Update = "updated"
+	Add = "added"
+}
 }
 resource "aws_iam_role" "iam_role" {
-  name = "tf-test-iam-dynamodb-role-%[1]s"
+name = "tf-test-iam-%[2]s-role-%[1]s"
 
-  assume_role_policy = <<EOF
+assume_role_policy = <<EOF
 {
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": "sts:AssumeRole",
-      "Principal": {
-        "Service": "dms.amazonaws.com"
-      },
-      "Effect": "Allow"
-    }
-  ]
+"Version": "2012-10-17",
+"Statement": [
+	{
+		"Action": "sts:AssumeRole",
+		"Principal": {
+			"Service": "dms.amazonaws.com"
+		},
+		"Effect": "Allow"
+	}
+]
 }
 EOF
 }
 
-resource "aws_iam_role_policy" "dms_dynamodb_access" {
-  name = "tf-test-iam-dynamodb-role-policy-%[1]s"
-  role = "${aws_iam_role.iam_role.name}"
+resource "aws_iam_role_policy" "dms_%[2]s_access" {
+name = "tf-test-iam-%[2]s-role-policy-%[1]s"
+role = "${aws_iam_role.iam_role.name}"
 
-  policy = <<EOF
+policy = <<EOF
 {
 "Version": "2012-10-17",
 "Statement": [
 {
-    "Effect": "Allow",
-    "Action": [
-        "dynamodb:PutItem",
-        "dynamodb:CreateTable",
-        "dynamodb:DescribeTable",
-        "dynamodb:DeleteTable",
-        "dynamodb:DeleteItem",
-        "dynamodb:ListTables"
-    ],
-    "Resource": "*"
+	"Effect": "Allow",
+	"Action": [
+			%[3]s
+	],
+	"Resource": "*"
 }
 ]
 }
 EOF
 }
-`, randId)
+`, randId, engineName, actionList)
+}
+
+func dmsEndpointDynamoDbConfig(randId string) string {
```

Comment:
Why don't we just statically assign all the `dynamodb:*` and `s3:*` permissions?